packages/artifact/src/internal-http-manager.ts

```diff
@@ -0,0 +1,36 @@
+import {HttpClient} from '@actions/http-client/index'
+import {createHttpClient} from './internal-utils'
+
+/**
+ * Used for managing all http Connections during either upload and download in order to limit the number of tcp connections created
+ * Separate clients for download and upload are used so that there are no conflicts if both are used at the same time
+ */
+export abstract class HttpManager {
```

Comment:
Neat! Is this pattern used in some of our other actions?

Reply:
I haven't seen this pattern used in any of our other actions, came up with this myself.

Using design patterns for the sake of using design patterns is never a good thing, but I thought this was as good use case for the singleton to ensure there is only ever class managing http connections